src/providers/ruby.rs

```diff
@@ -183,6 +187,8 @@ impl RubyProvider {
         ]);
 
         if self.is_rails_app(app) {
+            env_vars.insert("RAILS_ENV".to_string(), "production".to_string());
+            env_vars.insert("SECRET_KEY_BASE".to_string(), "1".to_string());
```

Comment:
This is required for asset compilation in production. I assume this will be overriden with the env defined by Railway in their UI

Reply:
Forgive me as I am not that familiar with deploying Ruby apps. But is it common for this env var to have a default value? Seems like it could be a security risk with someone forgetting to override this value when deploying to production.

Reply:
When precompiling assets a `SECRET_KEY_BASE` needs to be present. It might not be a good thing to not remember changing.

However we could inline it in the `bundle exec rails assets:precompile` call ?

Reply:
Maybe it should be possible to add build time environment variables, but that might increase complexity of nixpacks

Reply:
Seems that Heroku just inject it also https://github.com/heroku/buildpacks-ruby/blob/d988c4371cc5aef11e1c7375fad4f7ed87475de8/buildpacks/ruby/src/layers/bundle_install_layer.rs#L435

Reply:
What you linked is just a test. For prod it looks like they generate a random one https://github.com/heroku/buildpacks-ruby/blob/d988c4371cc5aef11e1c7375fad4f7ed87475de8/buildpacks/ruby/src/steps/default_env.rs#L51

I think it would be a good idea for Nixpacks to do the same. Thoughts?

Reply:
Ohh yeah

It seems it inserts a random one for the first build, but then uses an existing one otherwise. Is this possible for nixpacks? To generate an ENV and re-use it for the next build?

Reply:
At the moment, it is not possible. Nixpacks builds each image isolated. We would have to return some information that is saved in the host platform. In the meantime, I think we should remove the automatic `SECRET_KEY_BASE` variable.